src/overlays/actors/ovl_En_Ge3/z_en_ge3.c

```diff
@@ -28,49 +46,386 @@ const ActorInit En_Ge3_InitVars = {
+            play->transitionTrigger = TRANS_TRIGGER_START;
+            play->transitionType = TRANS_TYPE_38;
+            gSaveContext.save.weekEventReg[80] &= (u8)~8;
+        }
+    }
+}
+
+void EnGe3_AveilsChamberIdle(EnGe3* this, PlayState* play) {
+    if (SkelAnime_Update(&this->skelAnime) && (this->csAction == GERUDO_AVEIL_CSACTION_INSTRUCT)) {
+        EnGe3_ChangeAnim(this, GERUDO_AVEIL_ANIM_WAIT, ANIMMODE_ONCE, 0.0f);
+    }
+
+    if (Cutscene_CheckActorAction(play, 108)) {
+        s16 csAction = play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, 108)]->action;
+
+        if (this->csAction != GERUDO_AVEIL_CSACTION_BEEHIVE_RUN_AWAY) {
+            Cutscene_ActorTranslateAndYaw(&this->picto.actor, play, Cutscene_GetActorActionIndex(play, 108));
+        }
+
+        if (this->csAction != csAction) {
+            this->csAction = csAction;
+            switch (this->csAction) {
+                // Both cutscenes
+                case GERUDO_AVEIL_CSACTION_WAIT:
+                    EnGe3_ChangeAnim(this, GERUDO_AVEIL_ANIM_HAND_ON_HIP_WAIT, ANIMMODE_LOOP, 0.0f);
+                    break;
+
+                // Instructing expedition leader cutscene
+                case GERUDO_AVEIL_CSACTION_PAUSE:
+                    EnGe3_ChangeAnim(this, GERUDO_AVEIL_ANIM_TURN_HEAD, ANIMMODE_ONCE, 0.0f);
+                    this->skelAnime.playSpeed = 0.0f;
+                    break;
+
+                case GERUDO_AVEIL_CSACTION_INSTRUCT:
+                    EnGe3_ChangeAnim(this, GERUDO_AVEIL_ANIM_TURN_HEAD, ANIMMODE_ONCE, 0.0f);
+                    break;
+
+                case GERUDO_AVEIL_CSACTION_DEMAND:
+                    EnGe3_ChangeAnim(this, GERUDO_AVEIL_ANIM_DEMAND, ANIMMODE_LOOP, 0.0f);
+                    break;
+
+                case GERUDO_AVEIL_CSACTION_DISMISS:
+                    EnGe3_ChangeAnim(this, GERUDO_AVEIL_ANIM_DISMISS, ANIMMODE_ONCE, 0.0f);
+                    break;
+
+                case GERUDO_AVEIL_CSACTION_BEG:
+                    EnGe3_ChangeAnim(this, GERUDO_AVEIL_ANIM_BEG, ANIMMODE_LOOP, 0.0f);
+                    break;
+
+                // Fleeing from bees cutscene
+                case GERUDO_AVEIL_CSACTION_BEEHIVE_RUN_AWAY:
+                    EnGe3_ChangeAnim(this, GERUDO_AVEIL_ANIM_RUN_AWAY, ANIMMODE_LOOP, 0.0f);
+                    this->picto.actor.speedXZ = 5.0f;
+                    this->screamTimer = (s32)(Rand_ZeroFloat(10.0f) + 20.0f);
+                    break;
+
+                case GERUDO_AVEIL_CSACTION_BEEHIVE_VANISH:
+                    Actor_MarkForDeath(&this->picto.actor);
+                    break;
+            }
+
+            SkelAnime_Update(&this->skelAnime);
+        }
+    } else if ((this->picto.actor.xzDistToPlayer < 700.0f) && (fabsf(this->picto.actor.playerHeightRel) < 100.0f) &&
+               !Play_InCsMode(play)) {
+        func_800B7298(play, &this->picto.actor, 0x1A);
```

Comment:
```suggestion
        func_800B7298(play, &this->picto.actor, PLAYER_CSMODE_26);
```
Something I noticed in ge2